internal/config_extractor.go

```diff
@@ -0,0 +1,55 @@
+package internal
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	metadata "github.com/checkpoint-restore/checkpointctl/lib"
+)
+
+func ExtractConfigDump(checkpointPath string) (namespace, pod, container string, timestamp time.Time, err error) {
+	tempDir, err := os.MkdirTemp("", "extracted-checkpoint")
+	if err != nil {
+		return "", "", "", time.Time{}, err
+	}
+	defer os.RemoveAll(tempDir)
+
+	filesToExtract := []string{"config.dump"}
+	if err := UntarFiles(checkpointPath, tempDir, filesToExtract); err != nil {
+		log.Printf("Error extracting files from archive %s: %v\n", checkpointPath, err)
+		return "", "", "", time.Time{}, err
+	}
+
+	configDumpPath := filepath.Join(tempDir, "config.dump")
+	content, err := os.ReadFile(configDumpPath)
+	if err != nil {
+		log.Printf("Error reading config.dump file: %v\n", err)
+		return "", "", "", time.Time{}, err
+	}
+
+	return extractConfigDumpContent(content)
+}
+
+func extractConfigDumpContent(content []byte) (namespace, pod, container string, timestamp time.Time, err error) {
+	var c metadata.ContainerConfig
+	if err := json.Unmarshal(content, &c); err != nil {
+		return "", "", "", time.Time{}, err
+	}
+
+	parts := strings.Split(c.Name, "_")
```

Comment:
I would prefer to avoid string parsing. This information is available from the file spec.dump in the archive. You could then also add a field which container engine created the checkpoint. Currently it is only CRI-O for Kubernetes, but there is a PR for containerd open (for a long time).

In the annotations of spec.dump there is the information in io.kubernetes.cri-o.Labels. Take a look at https://github.com/cri-o/cri-o/blob/main/server/container_restore.go#L162

Reply:
Got it. Thanks @adrianreber for the information, I should also check the spec.dump. Sorry for the inconvenience.